### PR TITLE
Audio Spectrogram and Augment support

### PR DESCRIPTION
**Pull request by yongtang**
This PR adds initial audio spectrogram and augment support, to enrich the audio processing in tfio. APIs are placed under `tfio.experimental.audio`:

- `tfio.experimental.audio.spectrogram` to convert audio waveform to spectrogram (this is a a fused wrapper of `tf.signal.stft`)
- `tfio.experimental.audio.melscale` to allow creation of mel_spectrogram
- `tfio.experimental.audio.dbscale` to convert to dbscale for spectrogram
- `tfio.experimental.audio.freq_mask` to apply masking to a spectrogram in the freq domain
- `tfio.experimental.audio.time_mask` to apply masking to a spectrogram in the time domain

More details about freq/time masking are in [reference](https://arxiv.org/abs/1904.08779).

There are also discussions in #839 and implementations in DeepSpeech.

All implementations are python based, which helps gradient.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: tensorflow_io/core/python/experimental/audio_ops.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+import numpy as np
+
+import tensorflow as tf
+
+
+def spectrogram(input, nfft, window, stride, name=None):
+    """
+    Create spectrogram from audio.
+
+    TODO: Support audio with channel > 1.
```
Comment (terrytangyuan) [inline tensorflow_io/core/python/experimental/audio_ops.py:26]: Move TODO out of docstring?

File: tensorflow_io/core/python/experimental/audio_ops.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+    Args:
+      input: An 1-D audio signal Tensor.
+      nfft: Size of FFT.
+      window: Size of sindow.
```
Comment (terrytangyuan) [inline tensorflow_io/core/python/experimental/audio_ops.py:31]: sindow -> window

File: tensorflow_io/core/python/experimental/audio_ops.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    log_spec = 10.0 * (tf.math.log(power) - tf.math.log(10.0))
+    log_spec = tf.math.maximum(log_spec, tf.math.reduce_max(log_spec) - top_db)
+    return log_spec
+
+
+def freq_mask(input, param, name=None):
+    """
+    Apply masking to a spectrogram in the freq domain.
+
+    TODO: Support audio with channel > 1.
```
Comment (terrytangyuan) [inline tensorflow_io/core/python/experimental/audio_ops.py:102]: Same here

File: tests/test_audio_ops_eager.py [hunk 4]
```diff
@@ -805,3 +805,56 @@ def test_encode_mp3_mono():
...
+    rate = 10000
+    mels = 128
+    fmin = 0
+    fmax = 5000
+    mel_spectrogram = tfio.experimental.audio.melscale(
+        spectrogram, rate=rate, mels=mels, fmin=fmin, fmax=fmax
+    )
+
+    # TODO: assert content of mel_spectrogram
+    assert mel_spectrogram.shape == [29, 128]
```
Comment (terrytangyuan) [inline tests/test_audio_ops_eager.py:843]: Reuse the constants defined above

**Comment by 8bitmp3** [comment]
Hi @yongtang 👋 Thank you for adding spectrogram support. Do you by any chance have any plans to move `tfio.experimental.audio` or `tfio.experimental.audio.spectrogram` out of experimental? cc @mihaimaruseac @lamberta @MarkDaoust

**Comment by yongtang** [comment]
@8bitmp3 I created an issue in #1330. Let's track the progress there.

**Pull request merged**